### PR TITLE
FAV-16 allow the user to locally store advanced schema

### PR DESCRIPTION
**Pull request by joelplourde3**
- Rename FhavroConverter -> Fhavro, keep it simple because I've noticed that "Converter" does not really make sense anymore since the static class does a bit more than just convert and I want it to be user-friendly
- Implemented that Advanced schema are loaded from the user local repository rather than from the resources itself. In other words, it didnt make sense to packaged Advanced schema in the library itself because it is heavily dependent on how the "user" will use it and by user I mean the export etl as of right now.
- Added a couple of unit tests to validate this behaviour.

UPDATE:
- Added some javadoc for Fhavro. The api-docs looks swell :)

### CODE REVIEW

**Review by jecos — CHANGES_REQUESTED** [review]
Change version in pom.xml (0.10.0-SNAPSHOT)

File: src/main/java/bio/ferlab/fhir/Fhavro.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+     * @param schemaName The name of the schema.
+     * @param schemaMode The Schema Mode dictating where to search
+     * @return The Schema
+     */
+    @NotNull
+    public static Schema loadSchema(String schemaName, SchemaMode schemaMode) {
+        schemaName = ConverterUtils.formatSchemaName(schemaName);
+
+        try {
+            InputStream inputStream;
```
Comment (evans-g-crsj) [inline src/main/java/bio/ferlab/fhir/Fhavro.java:93]: `InputStream` must be closed explicitly or with a try-with-resources

File: src/main/java/bio/ferlab/fhir/Fhavro.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+    /**
+     * Load a Profile from an InputStream
+     *
+     * @param inputStream The InputStream where to load the Profile from.
+     * @return The Structure Definition of the Profile
+     */
+    @NotNull
+    public static StructureDefinition loadProfile(InputStream inputStream) {
+        try {
```
Comment (evans-g-crsj) [inline src/main/java/bio/ferlab/fhir/Fhavro.java:154]: stream must be closed

File: src/main/java/bio/ferlab/fhir/schema/definition/SchemaDefinition.java [hunk 3]
```diff
@@ -30,10 +30,10 @@ public SchemaDefinition(String schemaName) {
     public SchemaDefinition(String schemaName, String profileName, String[] extensionNames) {
         this(schemaName);
         if (StringUtils.isNotBlank(profileName)) {
-            setProfile(FhavroConverter.loadProfile(profileName));
+            setProfile(Fhavro.loadProfile(profileName));
             if (extensionNames != null && extensionNames.length > 0) {
```
Comment (evans-g-crsj) [inline src/main/java/bio/ferlab/fhir/schema/definition/SchemaDefinition.java:34]: (Optional): if CollectionUtils is available you could use the same pattern as with Strings at line 32 => CollectionUtils.isNotEmpty

**Review by evans-g-crsj — CHANGES_REQUESTED** [review]

File: src/main/java/bio/ferlab/fhir/Fhavro.java (hunk 1 above)
Comment (jecos) [reply src/main/java/bio/ferlab/fhir/Fhavro.java:93]: https://www.baeldung.com/java-stream-close

Comment (joelplourde3) [reply src/main/java/bio/ferlab/fhir/Fhavro.java:93]: Good point, this makes me question my sanity.

**Review by jecos — APPROVED** [review]

**Pull request merged**